examples/cargo_without_cmake/build.rs

```diff
@@ -23,5 +29,40 @@ fn main() {
             println!("cargo:rerun-if-changed=src/cpp/run.cpp");
         })
         .build();
+
+    // This fails to link with GNU ld, which is the default on most Linux distributions, so use lld or mold instead.
+    #[cfg(unix)]
+    {
+        let flags = env::var("CARGO_ENCODED_RUSTFLAGS").unwrap();
+        // Don't override custom flags
+        if !flags.contains("-fuse-ld") {
+            let linker_is_gnu = str::from_utf8(
+                &command_version_output("ld")
```

Comment:
as discussed lets put a comment here about how on macOS `ld` is actually `ldd` hence why this check is looking at the version output. As this wasn't obvious to me why this check was doing this.